### ISSUE TITLE
Public visible repository even with REQUIRE_SIGNIN_VIEW set to true (e.g. for Gitea Actions)

### ISSUE THREAD

**Issue by TheMasterFX** [go-gitea/gitea#28187]
### Feature Description

It would be good if Gitea is able to have one or more repositories or organisations that could be accessed without login, even if REQUIRE_SIGNIN_VIEW is set to true. It does not necessarily have to be visible in the web interface, but at least clonable if the URL is known.

Background:
In our company we have regulations that prohibit making internally created code visible to unauthenticated employees (need to know). However, since we do not have a direct internet connection from our servers, and we still want to use Gitea actions that are hosted in the Gitea instance itself (e.g. https://ourgitea.example.com/actions/....). But if we want to use it now we have to work with secrets or deploy keys with absolute URLs for actions when we create a CI/CD workflow, which is very unfortunate for portability.
Maybe I'm just making a mistake in thinking or there are already other solutions for this use case. Maybe it is even the right approach to create a special "gitea" user with permissions to all repositories and to work with access tokens which are then entered in the DEFAULT_ACTIONS_URL.

**Comment by lunny** [comment]
I think Actions repositories could be public/limited but not private so that every login user could visit it?

**Comment by TheMasterFX** [comment]
Yes but if REQUIRE_SIGNIN_VIEW is enabled even public can't be accessed without account

**Comment by Shuenhoy** [comment]
We found a workaround for this problem.
We create a special account that is `Restricted` (i.e it cannot access anything unless it is granted the permission explicitly).
Then we create a sub site in our caddy proxy `pub.git.tld`, where a readonly PAT of that account is passed to gitea.
```caddyfile
handle @gitpub {
reverse_proxy server:3000 {
	header_up Authorization {env.GITEA_PUB_SECRET}
}
}
```

This generally generally works well, but with two limitations:
* For actions, we have to write `use https://pub.git.tld/actions/xxxx`.
* This does not work for the packages of a private account, as we cannot grant the access of a private account to another account.

**Comment by bioinformatist** [comment]
@lunny We also met the problem. Can we add such a feature for supporting it?

**Comment by adriensamson** [comment]
Hello, I came across this issue today and I have found another workaround: requiring login only for explore.
With this config, anonymous users can view and clone public repos if (and only if) they know the URL.
```
[service]
REQUIRE_SIGNIN_VIEW = false
[service.explore]
REQUIRE_SIGNIN_VIEW = true
```

**Comment by San3-Cod3** [comment]
Not sure if correct place, but there is a serious privacy issue I am seeing with Gitea.
e.g.
https://gitea.com/USERNAME/RepoName/commit/fjh55gjufjrj575yjgjdjed8756jgfjjhkb7th (random string in example).

If making a mistake in a commit to your own Repo as a free cloud storage user, and you use CLI to squash, reword, rebase etc.. and fully clean the Repo to a single root commit without old history beyond your intention, the USER Public Activity is still linking to all these commits containing unwanted or sensitive data that YOU; the USER previously contributed over the account's lifetime, and even setting your Profile to limited, or private, these URLs remain active with sensitive information in them and can be seen by anyone with the URL, both with an account and without one.  These are all publicly visible, and I have a huge issue with this. 

I set my Activity to private, but these URLs are still active linking to history that isn't there anymore, and even if fully cleaning Repo, rebasing or even deleting it and its tags and pushing empty data back to origin main for Repo and tags, or even deleting the entire account, these URLs are still active with the sensitive information...

Is there a cleanup routine for this, is there a solution to this? I do not like this at all.

Just allow to us copy these URLs to an Area within our Account Settings and nuke them entirely.

Having a private Public Activity area does not stop this info from existing completely publicly...

How to purge these old and unrelated commits and History that aren't even associated to the Repo anymore from USER Activity completely?

I understand that it might need to exist for a time, related to auditing, but it does need to go; and quickly, with an option to expedite the process in Settings.
